packages/lesswrong/server/scripts/convertAllPostsToEAEmojis.ts

```diff
@@ -0,0 +1,123 @@
+/* eslint-disable no-console */
+
+import { getSqlClientOrThrow } from "../../lib/sql/sqlClient";
+import { pgPromiseLib } from "../sqlConnection";
+import { Globals } from "../vulcan-lib";
+
+type PostVotingSystem = Pick<DbPost, "_id" | "votingSystem">;
+type VoteType = Pick<DbVote, "documentId" | "extendedVoteType">;
+type VoteCounts = {agree: number, disagree: number};
+
+const convertDefaultVotingSystemToEAEmojis = async (postId: string) => {
+  await getSqlClientOrThrow().none(`
+    UPDATE "Posts"
+    SET "votingSystem" = 'eaEmojis'
+    WHERE "_id" = $1
+  `, [postId]);
+}
+
+const convertTwoAxisVotingSystemToEAEmojis = async (postId: string) => {
+  await getSqlClientOrThrow().tx(async (db) => {
+    const votes: VoteType[] = await db.any(`
+      UPDATE "Votes" AS v
+      SET "extendedVoteType" = CASE v."extendedVoteType"->>'agreement'
+        WHEN 'smallUpvote' THEN '{"agree":true}'::JSONB
+        WHEN 'bigUpvote' THEN '{"agree":true}'::JSONB
+        WHEN 'smallDownvote' THEN '{"disagree":true}'::JSONB
+        WHEN 'bigDownvote' THEN '{"disagree":true}'::JSONB
+        ELSE NULL
+        END
+      FROM "Comments" AS c
+      WHERE
+        c."_id" = v."documentId" AND
+        v."collectionName" = 'Comments' AND
+        c."postId" = $1 AND
+        v."cancelled" IS NOT TRUE AND
+        v."isUnvote" IS NOT TRUE AND
+        v."extendedVoteType"->>'agreement' IS NOT NULL
+      RETURNING "documentId", "extendedVoteType"
+    `, [postId]);
+
+    const voteCounts: Record<string, VoteCounts> = {};
+    for (const {documentId, extendedVoteType} of votes) {
+      if (!extendedVoteType) {
+        continue;
+      }
+      if (!voteCounts[documentId]) {
+        voteCounts[documentId] = {agree: 0, disagree: 0};
+      }
+      if (extendedVoteType.agree) {
+        voteCounts[documentId].agree++;
+      } else if (extendedVoteType.disagree) {
+        voteCounts[documentId].disagree++;
+      }
+    }
+
+    const commentIdsToUpdate = Object.keys(voteCounts);
+    const queries: {query: string, values: unknown[]}[] = [];
+    for (const commentId of commentIdsToUpdate) {
+      queries.push({
+        query: `
+          UPDATE "Comments"
+          SET "extendedScore" = "extendedScore" || $2::JSONB
+          WHERE "_id" = $1
+        `,
+        values: [commentId, voteCounts[commentId]],
+      });
+    }
+
+    queries.push({
+      query: `
+        UPDATE "Posts"
+        SET "votingSystem" = 'eaEmojis'
+        WHERE "_id" = $1
+      `,
+      values: [postId],
+    });
+
+    const concatenatedQuery = pgPromiseLib.helpers.concat(queries);
+    await db.multi(concatenatedQuery);
+  });
+}
+
+const convertAllPostsToEAEmojis = async () => {
+  console.log("Converting voting systems to EA emojis...");
+  const posts: PostVotingSystem[] = await getSqlClientOrThrow().any(`
+    SELECT "_id", "votingSystem"
+    FROM "Posts"
+    WHERE "votingSystem" <> 'eaEmojis'
+  `);
+
+  for (let i = 0; i < posts.length; i++) {
```

Comment:
Roughly how long this is expected to take?

Reply:
On my internet connection (which is pretty bad) I can do 2-3 posts per second and there's about 32000 posts than need to be converted, so about 3 hours. Because each post is done in a transaction it's totally safe to stop and restart the process at any point if needed.